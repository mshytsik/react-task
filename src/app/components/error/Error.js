import "./Error.scss";

const Error = () => {
  return <p className="error-text">Oops, there is no page here!</p>;
};

export default Error;
