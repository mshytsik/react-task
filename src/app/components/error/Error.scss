@import "../../style/base/index";

.error-text {
  margin: 60px 0 20px;
  font-weight: 700;
  font-size: 32px;
  line-height: 40px;
  color: $primary-color;
  text-align: center;

  @include media-breakpoint-down("md") {
    margin-top: 30px;
    font-size: 24px;
    line-height: 32px;
  }
}
