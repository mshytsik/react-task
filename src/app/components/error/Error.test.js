import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import Error from "./Error";

describe("test Error component", () => {
  test("is rendered correctly", () => {
    const component = renderer.create(<Error />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test("error message is correct", () => {
    const { getByText } = render(<Error />);
    const component = getByText("Oops, there is no page here!");
    expect(component).toBeInTheDocument();
  });

  test("has error class", () => {
    const { container } = render(<Error />);
    const component = container.getElementsByClassName("error-text");
    expect(component).toHaveLength(1);
  });
});
