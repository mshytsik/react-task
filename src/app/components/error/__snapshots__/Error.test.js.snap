// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Error component is rendered correctly 1`] = `
<p
  className="error-text"
>
  Oops, there is no page here!
</p>
`;
